include/VecCore/Backend/Implementation.h

```diff
@@ -284,6 +285,21 @@ VECCORE_FORCE_INLINE VECCORE_ATT_HOST_DEVICE constexpr Bool_s EarlyReturnAllowed
 #endif
 }
 
+// An extension of EarlyReturnAllowed(), and new name is a reminder that it can be used for other purposes as well
+template <typename T, size_t simdSize = 1>
+VECCORE_FORCE_INLINE
+VECCORE_ATT_HOST_DEVICE
+constexpr Bool_s SIMDsizeUpTo()
```

Comment:
>  I personally think that using EarlyReturnAllowed() && VectorSize<T>() >= n directly in code is the best thing. 

Well this leads to long line of code:
```if (vecCore::SIMDsizeUpTo<Real_v>() && vecCore::MaskFull(done)) return;```
becomes
```if (vecCore::EarlyReturnAllowed() && vecCore::VectorSize<T>() >= n && vecCore::MaskFull(done)) return;```
where most of the text is 'technical' hiding a bit the algorithm part ( ```MaskFull(done)``` )
doing using namespace vecCore, helps a bit:
```if (EarlyReturnAllowed() && VectorSize<T>() <= 1 && MaskFull(done)) return;```
but still ...